### PR TITLE
Updates to KFP End-to-end tutorial (Issue #860)

### PR DESCRIPTION
**Pull request by OfficePop**
In response to [kubeflow/website issue 860](https://github.com/kubeflow/website/issues/860), I have made the following edits to the KFP end-to-end GCP tutorial:

1. Updated with recommended kubeflow version from 0.4.1 to 0.5.1 with an html tag (html page has also been added in this PR). In the future, this can be handled by updating the embedded html page instead of directly updating the tutorial.

1. Revised the overview to recommend the user start with Cloud Shell instead of installing Cloud SDK. This reduces the steps the user needs to take and eliminates a potential chokepoint from errors installing or configuring a local copy of the SDK.

1. Removed the instructions to install kubectl locally, as it is already present in Cloud Shell. References to the local copy of kubectl were revised as well. After discussion with [joeliedtke](https://github.com/joeliedtke) it was agreed to retain the “handy environment variables” procedure when it was identified that it had dependencies later in the tutorially (mainly the user selecting their region/zone).

1. Instructions for granting user access were added to the “Set up your GCP account and SDK” procedure, along with the required roles and a reference link to the full procedure.

1. The commands apt-get install -y and wget bzip2 were removed from the “Set up Python” procedure.

1. A note was added at the end of the “Run the Pipeline” procedure to help the user locate the data for the trained model.

The only item that was not addressed from the original issue was item 6. This requires more familiarity with all aspects of the tutorial than I can provide at this time, and I recommend it be further fleshed out in another issue.

Also, I would suggest updating the screenshot in the “Deploy Kubeflow” procedure. The version number reads “0.4.1”, which out of date. If possible it would be preferable to update it to something generic like and reduce future maintenance.

This pull request is a copy of a [previous PR](https://github.com/kubeflow/website/pull/890) that had CLA and merging conflicts. Should be a clean start.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/921)

### CODE REVIEW

File: layouts/shortcodes/kubeflow-current-version.html [hunk 1]
```diff
@@ -0,0 +1,1136 @@
+
```
Comment (sarahmaddox) [inline layouts/shortcodes/kubeflow-current-version.html]: Oops I think something went wrong - maybe revert this change?

**Comment by sarahmaddox** [comment]
/ok-to-test

@joeliedtke when you're ready, please would you assign @DanSanche to do a tech review.

File: layouts/shortcodes/kubeflow-current-version.html (hunk 1 above)
Comment (OfficePop) [reply layouts/shortcodes/kubeflow-current-version.html]: Oops indeed! I've corrected it to just hold the version number as intended.

File: layouts/shortcodes/kubeflow-current-version.html [hunk 2]
```diff
@@ -0,0 +1 @@
+0.5.1
```
Comment (sarahmaddox) [inline layouts/shortcodes/kubeflow-current-version.html]: Ah, I see! Thanks for fixing the content. I have a question for @OfficePop and @joeliedtke: Is it the intention that this value refects the stable version or the latest version of Kubeflow? FYI the master version of the docs always reflects the latest version of Kubeflow, unless otherwise stated in the docs.

Another thing to check is whether you need `0.5.1` or `v0.5.1` for the purposes of your doc. (I haven't delved into the doc itself.)

The stable version is currently 0.5.1, and we usually refer to it as "v0.5.1". We already have a shortcode for this version, although it's out of date. I've raised PR https://github.com/kubeflow/website/pull/933 to update it. The shortcode is at `layouts/shortcodes/kf-stable-tag.html`.

If you're looking for the latest version, I've coincidentally just reviewed and commented in PR https://github.com/kubeflow/website/pull/928, which also needs the latest version. Since that PR and potentially your PR both need this shortcode, I've gone ahead and created the new shortcode in my own PR https://github.com/kubeflow/website/pull/933.

See the [Kubeflow releases](https://github.com/kubeflow/kubeflow/releases).

Comment (OfficePop) [reply layouts/shortcodes/kubeflow-current-version.html]: >  Is it the intention that this value refects the stable version or the latest version of Kubeflow?

@sarahmaddox yes, that is goal. We were trying to fix the fact that the live version of the page currently shows an older version (v0.4.1 I believe). Based on your comment I've removed the kubeflow current version tag and added the kf-stable-tag shortlink as well, so all should be well when both PRs are merged!

**Comment by sarahmaddox** [comment]
Thanks @OfficePop! 

/assign @DanSanche 

@DanSanche Please would you do a tech review of these updates to the KFP MNIST tutorial.

File: content/docs/pipelines/tutorials/pipelines-tutorial.md [hunk 3]
```diff
@@ -72,28 +71,20 @@ Follow these steps to set up your GCP environment:
...
-`kubectl` is the command-line tool for Kubernetes. `kubectl` is useful if you 
-want to interact with your Kubeflow cluster locally. If you decide not to use
-`kubectl`, you can skip the steps later in the tutorial that use the command.
-
-Run the following Cloud SDK command to install `kubectl`:
-
-```
-gcloud components install kubectl
-```
+* This guide assumes you want to manage your GCP environment in the [Cloud Shell][cloud-shell] environment rather than your own server. If you choose to use a local server instead, some of the components (such as Cloud SDK) must be manually installed.
```
Comment (joeliedtke) [inline content/docs/pipelines/tutorials/pipelines-tutorial.md]: Suggestion: We are recommending that they use Cloud Shell to save users the steps of installing Cloud SDK and kubectl. Let's rephrase this so users have a better understanding of the reason for the recommendation. How about something like this?

* This guide uses [Cloud Shell][cloud-shell] to manage your GCP environment, to save you the steps of installing [Cloud SDK][gcloud-install] and [kubectl][kubectl-install]. 

[kubectl-install]: https://kubernetes.io/docs/tasks/tools/install-kubectl/
[gcloud-install]: https://cloud.google.com/sdk/

File: content/docs/pipelines/tutorials/pipelines-tutorial.md [hunk 4]
```diff
@@ -72,28 +71,20 @@ Follow these steps to set up your GCP environment:
 1. Select or create a project on the [GCP Console][gcp-console].
 1. Make sure that billing is enabled for your project. See the guide to
   [modifying a project's billing settings][billing-guide].
-1. Install the [Cloud SDK][cloud-sdk].
+1. Use Cloud Shell to grant your team access to Kubeflow by granting them **IAP-secured Web App User** roles:
```
Comment (joeliedtke) [inline content/docs/pipelines/tutorials/pipelines-tutorial.md]: My understanding is that users need the Project Owner and IAP-secured Web App User roles to use Kubeflow on GKE. Project Owner can only be assigned in Cloud Console, so it probably makes sense to have them use Cloud Console. How about something like this?

1. Use [Cloud Console to grant your team access][grant-access] to Kubeflow by granting them access to the following roles:

    *  **Project Owner**: To ensure that your team can access all of the resources used in this guide, grant your team access to the **Project Owner** role.
    * **IAP-secured Web App User**: This guide uses Cloud Identity-Aware Proxy (IAP) to secure access to your Kubeflow cluster. Your team must be members of the **IAP-secured Web App User** role to authenticate with the Kubeflow web application.
 
[grant-access]: https://cloud.google.com/iam/docs/granting-changing-revoking-access#using_the

File: content/docs/pipelines/tutorials/pipelines-tutorial.md (hunk 3 above)
Comment (OfficePop) [reply content/docs/pipelines/tutorials/pipelines-tutorial.md]: Agreed and appreciated! I have updated the text accordingly.

File: content/docs/pipelines/tutorials/pipelines-tutorial.md (hunk 4 above)
Comment (OfficePop) [reply content/docs/pipelines/tutorials/pipelines-tutorial.md]: That makes sense, I appreciate your pointing this out. I have updated the text with your suggestion and some mild edits as follows:

  1. Use [Cloud console to grant your team access](https://cloud.google.com/iam/docs/granting-changing-revoking-access#using_the) to Kubeflow _by assigning them the following roles_:
     
      - **Project Owner**: _Ensures that your team can access all of the resources used in this guide._
      - **IAP-secured Web App User**: This guide uses Cloud Identity-Aware Proxy (IAP) to secure access to your Kubeflow cluster. Your team must be members of the **IAP-secured Web App User** role to authenticate with the Kubeflow web application.

Thanks!

File: content/docs/pipelines/tutorials/pipelines-tutorial.md [hunk 5]
```diff
@@ -610,4 +602,5 @@ SDK](/docs/pipelines/sdk/sdk-overview/).
 [gsutil-mb]: https://cloud.google.com/storage/docs/gsutil/commands/mb
 [gsutil-acl-ch]: https://cloud.google.com/storage/docs/gsutil/commands/acl#ch
 
-
+[granting-changing-revoking-access]: https://cloud.google.com/iam/docs/granting-changing-revoking-access#quick_updates_using_the_gcloud_command-line_tool
```
Comment (joeliedtke) [inline content/docs/pipelines/tutorials/pipelines-tutorial.md]: Sorry, this change wasn't called out as well as I would have liked in my previous review. Would you please update this link to be the following? 

https://cloud.google.com/iam/docs/granting-changing-revoking-access#using_the

This will take users to the instructions for updating IAM settings in the Cloud Console instead of the Cloud Shell.

Comment (OfficePop) [reply content/docs/pipelines/tutorials/pipelines-tutorial.md]: Got it, updated.

**Comment by joeliedtke** [comment]
/hold 

Hold for technical review.

File: content/docs/pipelines/tutorials/pipelines-tutorial.md [hunk 6]
```diff
@@ -158,7 +146,7 @@ Deploy Kubeflow on GCP:
 1. (Optional) If you want to examine your cluster while waiting for the UI to
    be available, you can use `kubectl` to connect to your cluster:
 
-  * Connect your local `kubectl` session to the cluster:
+  * Connect your Cloud Shell session to the cluster:
```
Comment (DanSanche) [inline content/docs/pipelines/tutorials/pipelines-tutorial.md]: nit: it might make sense to still reference kubectl here (maybe "Connect your Cloud Shell kubectl session"?). Users may not understand how this connection works

**Comment by sarahmaddox** [comment]
Thanks all for the review and for this update! I'm going to remove the hold and let this change get merged now, as the only recommendation outstanding is a self-acknowledged "nit". I want to avoid merge conflicts if anyone else updates the docs in the meantime. 

If anyone wants to submit another PR to fix the "nit" please do. :)

Congrats @OfficePop! 

/assign
/hold cancel
/approve
